mysql-test/suite/rpl/t/rpl_rewrite_db_sys_vars.test

```diff
@@ -0,0 +1,103 @@
+-- source include/have_binlog_format_mixed.inc
+-- source include/master-slave.inc
+
+--echo #
+--echo # MDEV-15530 Variable replicate_rewrite_db cannot be found
+--echo #            in "show global variables"
+--echo #
+
+--echo # Create DBs and verify that slave has to be stopped before setting sys var
+
+connection slave;
+# These DBs will be rewrited from opt file
+create database replica_db1;
+create database y;
+# This DB will be rewrited from test case
+create database test_replica;
+SELECT @@GLOBAL.replicate_rewrite_db;
+
+--echo # Create DBs and tables on primary
+connection master;
+--enable_warnings
+create database primary_db1;
+create database x;
+use primary_db1;
+create table my_table (t int);
+insert into my_table values (2),(4);
+use x;
+create table my_table (t int);
+insert into my_table values (654),(532);
+connection master;
+--source include/save_master_gtid.inc
+
+--echo # Check replica
+--connection slave
+--source include/sync_with_master_gtid.inc
+--let $diff_tables=master:primary_db1.my_table,slave:replica_db1.my_table
+--source include/diff_tables.inc
+--let $diff_tables=master:x.my_table,slave:y.my_table
+--source include/diff_tables.inc
+SELECT @@GLOBAL.replicate_rewrite_db;
+show tables from replica_db1;
+select * from replica_db1.my_table;
+show tables from y;
+select * from y.my_table;
+
+--echo # Set replica sys var replicate_rewrite_db
+connection slave;
+--error ER_SLAVE_MUST_STOP
+SET @@GLOBAL.replicate_rewrite_db="test_master->test_replica";
```

Comment:
Here, the `x->y` rewrite rule should be deleted in favor of only `test_master->test_replica`, right? Can you extend this test to show that updates in database `x` are no longer carried out into `y`? E.g., update `my_table` on the master in `x` and use `include/wait_for_slave_sql_error.inc`

Reply:
Done!